### PR TITLE
First attempt at creating site/variable/file lookup table for temp WQP pulls

### PR DESCRIPTION
**Pull request by limnoliver**
This is complete through creating a lookup/task table. Strategy borrows from workflows of lake-temperature-model-prep and the NAWQA WQP pull work. Toy example included is 6 sites (in 6 different lakes), 4 of which we do not currently have data. Workflow is as follows:

- Input sites and variables of interest for which you want data. Variables includes multiple names for same thing in WQP (e.g., Temperature, water (deg F) and Temperature, water for variable "temperature"). This is flexible to allow multiple variables. 

- Combine the sites and variables and define as your "wants". Compare them to your "haves" -- which is the sites/variables you already have. Subtract the two and come up with your "needs".

- Inventory WQP based on your "needs", then partition your needs up into appropriately-sized WQP pulls via a task list/lookup table. (Note, the inventory is taking too long - is this because it's being done by site? See #1) The table lists all site/variable combinations in your "needs", and includes a task grouping column that can be used for file names later on.  

Dependency diagram of these steps:

0_config_diagram

### CODE REVIEW

File: .gitignore/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# History files
```
Comment (jordansread) [inline .gitignore/.gitignore:1]: I don't think this file should exist, right?

File: 0_config/src/wqp_utils.R [hunk 2]
```diff
@@ -0,0 +1,94 @@
+inventory_wqp <- function(wqp_needs, wqp_variables) {
```
Comment (jordansread) [inline 0_config/src/wqp_utils.R:16]: This is tricky - 

if the number of sites is small, it is faster to just get the data than it is to re-build the inventory. If the number of sites is large, we'll need the inventory to break up the queries logically. 

Either way, I don't recommend including the logic to skip the inventory, just surprised that the query takes so long (~1.5 mins for 4 sites).

File: 0_config/src/wqp_utils.R [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+      })
+    }))
+  })
+  message(sprintf('sample inventory complete, required %0.0f seconds', sample_time[['elapsed']]))
+  
+  return(samples)
+}
+
+partition_wqp_inventory <- function(wqp_inventory, wqp_nrecords_chunk) {
```
Comment (jordansread) [inline 0_config/src/wqp_utils.R]: I know I didn't do a good job of this on the other repo, but can we add
`#' @param {paramname} {info here}` for each argument in the function? Will help us in the future to know what the expectations/assumptions are for those values

**Review by jordansread — APPROVED** [review]

File: .gitignore/.gitignore (hunk 1 above)
Comment (limnoliver) [reply .gitignore/.gitignore:1]: .Rhistory does. I usually have this turned off in RStudio but have been working from a number of loaner machines over the past ~3 weeks so this file snuck in.

File: 0_config/src/wqp_utils.R (hunk 3 above)
Comment (limnoliver) [reply 0_config/src/wqp_utils.R]: documentation to all functions added

**Pull request merged**